src/utils/extractTestResults.ts

```diff
@@ -2,25 +2,22 @@
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import { TestActivity } from './testActivity';
-import { TestResultModel } from './testResult';
+import { TestResultModel, VehicleType } from './testResult';
 
-export const extractBillableTestResults = (record: TestResultModel): TestActivity[] => {
-  const testActivities: TestActivity[] = record.testTypes.map((testType) => ({
-    noOfAxles: record.noOfAxles,
-    testTypeStartTimestamp: record.testStartTimestamp,
-    testTypeEndTimestamp: record.testEndTimestamp,
-    testStationType: record.testStationType,
-    testCode: testType.testCode,
-    vin: record.vin,
-    vrm: record.vrm,
-    testStationPNumber: record.testStationPNumber,
-    testResult: testType.testResult,
-    certificateNumber: testType.certificateNumber,
-    testTypeName: testType.name,
-    vehicleType: record.vehicleType,
-    testerName: record.testerName,
-    testerStaffId: record.testerStaffId,
-    testResultId: record.testResultId,
-  }));
-  return testActivities;
-};
+export const extractBillableTestResults = (record: TestResultModel): TestActivity[] => record.testTypes.map((testType) => ({
+  noOfAxles: record.noOfAxles,
+  testTypeStartTimestamp: record.testStartTimestamp,
+  testTypeEndTimestamp: record.testEndTimestamp,
+  testStationType: record.testStationType,
+  testCode: testType.testCode,
+  vin: record.vin,
+  vrm: record.vehicleType === VehicleType.TRL ? record.trailerId : record.vrm,
```

Comment:
You can null coalesce on vrm as it isn't defined on trailers: `vrm: record.vrm ?? record.trailerId`

Reply:
just feels safer this way since there _could_ be a situation were you get some bad data and the trailer has a vrm.